### PR TITLE
Remove unmeaning logic

### PR DESCRIPTION
**Pull request by wenxuwan**
**What this PR does**:

Remove the unmeaning variables in RegistryDataListener and change unlock to defer.

### CODE REVIEW

File: registry/zookeeper/registry.go [hunk 1]
```diff
@@ -290,20 +290,17 @@ func (r *zkRegistry) getCloseListener(conf *common.URL) (*RegistryConfigurationL
 
 	var zkListener *RegistryConfigurationListener
 	r.dataListener.mutex.Lock()
+	defer r.dataListener.mutex.Unlock()
```
Comment (zouyx) [inline registry/zookeeper/registry.go:293]: If this lock use defer, will include [listenerLock](https://github.com/apache/dubbo-go/blob/234633932aab82913ede37c3f32c7a48a912b869/registry/zookeeper/registry.go#L309), is it right logic?

**Comment by AlexStocks** [comment]
wait for @CodingSinger's comment

**Comment by CodingSinger** [comment]
> wait for @CodingSinger's comment

看了下之前的pr，这里应该是忘了写了，应该在close以及替换的时候设置close为true

**Review by zouyx — COMMENTED** [review]
Make `checks` pass pls

File: registry/zookeeper/registry.go (hunk 1 above)
Comment (wenxuwan) [reply registry/zookeeper/registry.go]: No.....getListener use  defer too, so here may be cause dead lock.

**Comment by wenxuwan** [comment]
> Make `checks` pass pls

done

**Review by fangyincheng — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Review by zouyx — APPROVED** [review]

**Pull request merged**